src/libraries/Common/src/Interop/Windows/Netapi32/Interop.DsRoleGetPrimaryDomainInformation.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Netapi32
+    {
+        internal enum DSROLE_PRIMARY_DOMAIN_INFO_LEVEL
+        {
+            DsRolePrimaryDomainInfoBasic = 1,
+            DsRoleUpgradeStatus = 2,
+            DsRoleOperationState = 3,
+            DsRolePrimaryDomainInfoBasicEx = 4
+        }
+
+        [LibraryImport(Libraries.Netapi32, StringMarshalling = StringMarshalling.Utf16)]
+        internal static partial int DsRoleGetPrimaryDomainInformation(
+            [MarshalAs(UnmanagedType.LPTStr)] string lpServer,
```

Comment:
The definition of this parameter is `LPWCSTR` which matches UTF16. Is `LPTStr` overridden in old code?

Reply:
Except for removing the `EntryPoint` property, I didn't change anything here. The code generated for `LPTStr` vs `LPWStr` looks identical to me (both with and without the `StringMarshalling` property set). Should I just remove the `MarshalAs` attribute altogether here?

Reply:
Yes please. `LPTStr` looks wrong here. It seems to be overridden by `StringMarshalling.Utf16`.

Reply:
Done in f4dfe7f7271cea758098f065284cc55a8517f149

Reply:
Agreed we should match the Windows declarations. But @Jkotas when is LPWSTR different to LPTSTR in a managed app? UNICODE is effectively always defined. (Also, what practical effect of any on marshaling does the C for const in LPCWSTR have?)

Reply:
Ah - when CharSet is not Unicode, as already mentioned.